From 0000000000000000000000000000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hayanesuru <hayanesuru@outlook.jp>
Date: Wed, 8 Jul 2026 16:16:48 +0900
Subject: [PATCH] remove gui

SPDX-License-Identifier: Unlicense

This is free and unencumbered software released into the public domain.

Anyone is free to copy, modify, publish, use, compile, sell, or
distribute this software, either in source code form or as a compiled
binary, for any purpose, commercial or non-commercial, and by any
means.

In jurisdictions that recognize copyright laws, the author or authors
of this software dedicate any and all copyright interest in the
software to the public domain. We make this dedication for the benefit
of the public at large and to the detriment of our heirs and
successors. We intend this dedication to be an overt act of
relinquishment in perpetuity of all present and future rights to this
software under copyright law.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
OTHER DEALINGS IN THE SOFTWARE.

For more information, please refer to <https://unlicense.org>

diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 27dd0c7923c6dc4b79d9f7c31de623ab7f2b0cc1c2b7e08b973393957123e2d0..4484b3a437985fbd028d99714a9bf242b7b92a1f697ccca92be5714a09dd49e3 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -36,7 +36,7 @@ import net.minecraft.server.ServerInterface;
 import net.minecraft.server.ServerLinks;
 import net.minecraft.server.Services;
 import net.minecraft.server.WorldStem;
-import net.minecraft.server.gui.MinecraftServerGui;
+// import net.minecraft.server.gui.MinecraftServerGui; // Haya - remove gui
 import net.minecraft.server.jsonrpc.ManagementServer;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
@@ -81,7 +81,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
     private final RconConsoleSource rconConsoleSource;
     private @Nullable RconThread rconThread;
     private final DedicatedServerSettings settings;
-    private @Nullable MinecraftServerGui gui;
+    // private @Nullable MinecraftServerGui gui; // Haya - remove gui
     private final @Nullable ServerTextFilter serverTextFilter;
     private @Nullable RemoteSampleLogger tickTimeLogger;
     private boolean isTickTimeLoggingEnabled;
@@ -462,9 +462,11 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
             this.serverTextFilter.close();
         }
 
-        if (this.gui != null) {
-            this.gui.close();
-        }
+        // Haya start - remove gui
+        // if (this.gui != null) {
+        //     this.gui.close();
+        // }
+        // Haya end - remove gui
 
         if (this.rconThread != null) {
             this.rconThread.stop();
@@ -561,9 +563,11 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
     }
 
     public void showGui() {
-        if (this.gui == null) {
-            this.gui = MinecraftServerGui.showFrameFor(this);
-        }
+        // Haya start - remove gui
+        // if (this.gui == null) {
+        //     this.gui = MinecraftServerGui.showFrameFor(this);
+        // }
+        // Haya end - remove gui
     }
 
     public int spawnProtectionRadius() {
diff --git a/net/minecraft/server/gui/MinecraftServerGui.java b/net/minecraft/server/gui/MinecraftServerGui.java
deleted file mode 100644
index 5242f88eba9518549659f1f71df481e5b4035e8f8f1e116f130095466ca72dba..0000000000000000000000000000000000000000000000000000000000000000
--- a/net/minecraft/server/gui/MinecraftServerGui.java
+++ /dev/null
@@ -1,177 +0,0 @@
-package net.minecraft.server.gui;
-
-import com.google.common.collect.Lists;
-import com.mojang.logging.LogQueues;
-import com.mojang.logging.LogUtils;
-import java.awt.BorderLayout;
-import java.awt.Dimension;
-import java.awt.Font;
-import java.awt.event.FocusAdapter;
-import java.awt.event.FocusEvent;
-import java.awt.event.WindowAdapter;
-import java.awt.event.WindowEvent;
-import java.util.Collection;
-import java.util.concurrent.atomic.AtomicBoolean;
-import javax.swing.JComponent;
-import javax.swing.JFrame;
-import javax.swing.JList;
-import javax.swing.JPanel;
-import javax.swing.JScrollBar;
-import javax.swing.JScrollPane;
-import javax.swing.JTextArea;
-import javax.swing.JTextField;
-import javax.swing.SwingUtilities;
-import javax.swing.UIManager;
-import javax.swing.border.EtchedBorder;
-import javax.swing.border.TitledBorder;
-import javax.swing.text.BadLocationException;
-import javax.swing.text.Document;
-import net.minecraft.DefaultUncaughtExceptionHandler;
-import net.minecraft.server.dedicated.DedicatedServer;
-import org.slf4j.Logger;
-
-public class MinecraftServerGui extends JComponent {
-    private static final Font MONOSPACED = new Font("Monospaced", 0, 12);
-    private static final Logger LOGGER = LogUtils.getLogger();
-    private static final String TITLE = "Minecraft server";
-    private static final String SHUTDOWN_TITLE = "Minecraft server - shutting down!";
-    private final DedicatedServer server;
-    private Thread logAppenderThread;
-    private final Collection<Runnable> finalizers = Lists.newArrayList();
-    private final AtomicBoolean isClosing = new AtomicBoolean();
-
-    public static MinecraftServerGui showFrameFor(final DedicatedServer server) {
-        try {
-            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
-        } catch (Exception var3) {
-        }
-
-        final JFrame frame = new JFrame("Minecraft server");
-        final MinecraftServerGui gui = new MinecraftServerGui(server);
-        frame.setDefaultCloseOperation(2);
-        frame.add(gui);
-        frame.pack();
-        frame.setLocationRelativeTo(null);
-        frame.setVisible(true);
-        frame.addWindowListener(new WindowAdapter() {
-            @Override
-            public void windowClosing(final WindowEvent event) {
-                if (!gui.isClosing.getAndSet(true)) {
-                    frame.setTitle("Minecraft server - shutting down!");
-                    server.halt(true);
-                    gui.runFinalizers();
-                }
-            }
-        });
-        gui.addFinalizer(frame::dispose);
-        gui.start();
-        return gui;
-    }
-
-    private MinecraftServerGui(final DedicatedServer server) {
-        this.server = server;
-        this.setPreferredSize(new Dimension(854, 480));
-        this.setLayout(new BorderLayout());
-
-        try {
-            this.add(this.buildChatPanel(), "Center");
-            this.add(this.buildInfoPanel(), "West");
-        } catch (Exception e) {
-            LOGGER.error("Couldn't build server GUI", e);
-        }
-    }
-
-    public void addFinalizer(final Runnable finalizer) {
-        this.finalizers.add(finalizer);
-    }
-
-    private JComponent buildInfoPanel() {
-        JPanel panel = new JPanel(new BorderLayout());
-        StatsComponent comp = new StatsComponent(this.server);
-        this.finalizers.add(comp::close);
-        panel.add(comp, "North");
-        panel.add(this.buildPlayerPanel(), "Center");
-        panel.setBorder(new TitledBorder(new EtchedBorder(), "Stats"));
-        return panel;
-    }
-
-    private JComponent buildPlayerPanel() {
-        JList<?> playerList = new PlayerListComponent(this.server);
-        JScrollPane scrollPane = new JScrollPane(playerList, 22, 30);
-        scrollPane.setBorder(new TitledBorder(new EtchedBorder(), "Players"));
-        return scrollPane;
-    }
-
-    private JComponent buildChatPanel() {
-        JPanel panel = new JPanel(new BorderLayout());
-        JTextArea chatArea = new JTextArea();
-        JScrollPane scrollPane = new JScrollPane(chatArea, 22, 30);
-        chatArea.setEditable(false);
-        chatArea.setFont(MONOSPACED);
-        JTextField chatField = new JTextField();
-        chatField.addActionListener(event -> {
-            String text = chatField.getText().trim();
-            if (!text.isEmpty()) {
-                this.server.handleConsoleInput(text, this.server.createCommandSourceStack());
-            }
-
-            chatField.setText("");
-        });
-        chatArea.addFocusListener(new FocusAdapter() {
-            @Override
-            public void focusGained(final FocusEvent arg0) {
-            }
-        });
-        panel.add(scrollPane, "Center");
-        panel.add(chatField, "South");
-        panel.setBorder(new TitledBorder(new EtchedBorder(), "Log and chat"));
-        this.logAppenderThread = new Thread(() -> {
-            String line;
-            while ((line = LogQueues.getNextLogEvent("ServerGuiConsole")) != null) {
-                this.print(chatArea, scrollPane, line);
-            }
-        }, "Server log monitor");
-        this.logAppenderThread.setUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler(LOGGER));
-        this.logAppenderThread.setDaemon(true);
-        return panel;
-    }
-
-    public void start() {
-        this.logAppenderThread.start();
-    }
-
-    public void close() {
-        if (!this.isClosing.getAndSet(true)) {
-            this.runFinalizers();
-        }
-    }
-
-    private void runFinalizers() {
-        this.finalizers.forEach(Runnable::run);
-    }
-
-    public void print(final JTextArea console, final JScrollPane scrollPane, final String line) {
-        if (!SwingUtilities.isEventDispatchThread()) {
-            SwingUtilities.invokeLater(() -> {
-                this.print(console, scrollPane, line);
-            });
-        } else {
-            Document document = console.getDocument();
-            JScrollBar scrollBar = scrollPane.getVerticalScrollBar();
-            boolean shouldScroll = false;
-            if (scrollPane.getViewport().getView() == console) {
-                shouldScroll = scrollBar.getValue() + scrollBar.getSize().getHeight() + MONOSPACED.getSize() * 4
-                    > scrollBar.getMaximum();
-            }
-
-            try {
-                document.insertString(document.getLength(), line, null);
-            } catch (BadLocationException var8) {
-            }
-
-            if (shouldScroll) {
-                scrollBar.setValue(Integer.MAX_VALUE);
-            }
-        }
-    }
-}
